components/hab/src/error.rs

```diff
@@ -127,6 +128,13 @@ impl fmt::Display for Error {
             Error::HabitatCore(ref e) => format!("{}", e),
             Error::HandlebarsRenderError(ref e) => format!("{}", e),
             Error::IO(ref err) => format!("{}", err),
+            Error::JobGroupPromote(ref id, ref c) => {
+                format!(
+                    "Failed to promote job group ID {} to channel {} because the job group is not complete.",
```

Comment:
OK, so maybe say "because some packages have not yet been built".

What are the four reasons a project might not have succeeded? Are those listed anywhere that a user could access them?